libs/parallelism/algorithms/include/hpx/parallel/container_algorithms/uninitialized_copy.hpp

```diff
@@ -0,0 +1,505 @@
+    hpx::parallel::util::in_out_result<InIter, FwdIter> uninitialized_copy(
+        InIter first1, Sent1 last1, FwdIter first2, Sent2 last2);
+
+    /// Copies the elements in the range, defined by [first, last), to an
+    /// uninitialized memory area beginning at \a dest. If an exception is
+    /// thrown during the copy operation, the function has no effects.
+    ///
+    /// \note   Complexity: Performs exactly \a last - \a first assignments.
+    ///
+    /// \tparam ExPolicy    The type of the execution policy to use (deduced).
+    ///                     It describes the manner in which the execution
+    ///                     of the algorithm may be parallelized and the manner
+    ///                     in which it executes the assignments.
+    /// \tparam FwdIter1    The type of the source iterators used (deduced).
+    ///                     This iterator type must meet the requirements of an
+    ///                     input iterator.
+    /// \tparam Sent1       The type of the source sentinel (deduced). This
+    ///                     sentinel type must be a sentinel for InIter.
+    /// \tparam FwdIter2    The type of the iterator representing the
+    ///                     destination range (deduced).
+    ///                     This iterator type must meet the requirements of a
+    ///                     forward iterator.
+    /// \tparam Sent2       The type of the source sentinel (deduced). This
+    ///                     sentinel type must be a sentinel for InIter2.
+    ///
+    /// \param policy       The execution policy to use for the scheduling of
+    ///                     the iterations.
+    /// \param first1       Refers to the beginning of the sequence of elements
+    ///                     that will be copied from
+    /// \param last1        Refers to sentinel value denoting the end of the
+    ///                     sequence of elements the algorithm will be applied to.
+    /// \param first2       Refers to the beginning of the destination range.
+    /// \param last2        Refers to sentinel value denoting the end of the
+    ///                     second range the algorithm will be applied to.
+    ///
+    /// The assignments in the parallel \a uninitialized_copy algorithm invoked
+    /// with an execution policy object of type \a sequenced_policy
+    /// execute in sequential order in the calling thread.
+    ///
+    /// The assignments in the parallel \a uninitialized_copy algorithm invoked
+    /// with an execution policy object of type \a parallel_policy or
+    /// \a parallel_task_policy are permitted to execute in an
+    /// unordered fashion in unspecified threads, and indeterminately sequenced
+    /// within each thread.
+    ///
+    /// \returns  The \a uninitialized_copy algorithm returns a
+    ///           \a hpx::future<in_out_result<InIter, FwdIter>>, if the execution policy is of type
```

Comment:
Please limit line length to max 80 characters (90 characters is the hard limit).